boto/connection.py

```diff
@@ -796,27 +796,28 @@ def proxy_ssl(self, host=None, port=None):
         else:
             sock = socket.create_connection((self.proxy, int(self.proxy_port)))
         boto.log.debug("Proxy connection: CONNECT %s HTTP/1.0\r\n", host)
-        sock.sendall("CONNECT %s HTTP/1.0\r\n" % host)
-        sock.sendall("User-Agent: %s\r\n" % UserAgent)
+        sock.sendall(six.ensure_binary("CONNECT %s HTTP/1.0\r\n" % host))
+        sock.sendall(six.ensure_binary("User-Agent: %s\r\n" % UserAgent))
         if self.proxy_user and self.proxy_pass:
             for k, v in self.get_proxy_auth_header().items():
-                sock.sendall("%s: %s\r\n" % (k, v))
+                sock.sendall(six.ensure_binary("%s: %s\r\n" % (k, v)))
             # See discussion about this config option at
             # https://groups.google.com/forum/?fromgroups#!topic/boto-dev/teenFvOq2Cc
             if config.getbool('Boto', 'send_crlf_after_proxy_auth_headers', False):
-                sock.sendall("\r\n")
+                sock.sendall(six.ensure_binary("\r\n"))
         else:
-            sock.sendall("\r\n")
+            sock.sendall(six.ensure_binary("\r\n"))
         resp = http_client.HTTPResponse(sock, strict=True, debuglevel=self.debug)
         resp.begin()
 
         if resp.status != 200:
             # Fake a socket error, use a code that make it obvious it hasn't
             # been generated by the socket library
             raise socket.error(-71,
+                               six.ensure_binary(
                                "Error talking to HTTP proxy %s:%s: %s (%s)" %
```

Comment:
Nit: This should be indented another 4 lines now.

We can probably move these args back, as they're getting close to the char limit:

```
            raise socket.error(
                -71,
                six.ensure_binary(
                    "...
```

Reply:
Got it, fixed the spacing up a lil :)